### ISSUE TITLE
Your setup does not contains any table.

### ISSUE THREAD

**Issue by VagueGit** [Mimetis/Dotmim.Sync#475]
I upgraded both web server and client to DMS 0.6.2 and on the first sync got exception `Your setup does not contains any table.`. This is on a new SQL Server database syncing from a local SQLite database over http.

The process we use is
Create empty SQL Server database
DMS creates local SQLite database from SQL Server
Populate local SQLite database with test data
Sync

I rolled back to DMS 0.6.1-beta-0504 and now everything is working as normal

**Comment by Mimetis** [comment]
Are you sure you have the same code?
Are you sure you did not forget to pass the SyncSetup?
Can you share the server code that fail?

**Comment by Mimetis** [comment]
By the way, "Configuration has no table" is not a DMS known error.

**Comment by VagueGit** [comment]
There are no changes on either the client or the web server other than the upgrade to DMS 0.6.2. The server code is pretty much straight out of the DMS songbook except we get the database and list of tables at runtime

On the server
```
public SyncController(WebServerManager webServerManager) => this.webServerManager = webServerManager;

[HttpPost]
public async Task Post(string databaseName, string createSnapshot)
{
    try
    {
        if (string.IsNullOrWhiteSpace(databaseName))
            return;
        // remove junk from database name
        databaseName = String.Concat(databaseName.Where(ch => Char.IsLetterOrDigit(ch)));

        if (!string.IsNullOrWhiteSpace(createSnapshot))
            createSnapshot = String.Concat(createSnapshot.Where(ch => Char.IsLetterOrDigit(ch))).ToLower();

        // Get Orchestrator regarding the incoming scope name (from http context)
        WebServerOrchestrator orchestrator = webServerManager.GetOrchestrator(this.HttpContext);
        //bool upgrade = await orchestrator.NeedsToUpgradeAsync();
        //if (upgrade)
        //    await orchestrator.UpgradeAsync();

        ConfigureOrchestrator(orchestrator, databaseName);

        SynchronousProgress<ProgressArgs> progress = new SynchronousProgress<ProgressArgs>(pa => Debug.WriteLine($"{pa.Context.SyncStage}\t {pa.Message}"));

        // handle request
        await webServerManager.HandleRequestAsync(this.HttpContext, default, progress);

        if (createSnapshot == "true")
            await orchestrator.CreateSnapshotAsync();
    }
    catch (Exception ex)
    {
        await WebServerManager.WriteExceptionAsync(this.HttpContext.Response, ex);
    }
}

private void ConfigureOrchestrator(WebServerOrchestrator orchestrator, string databaseName)
{
    if (string.IsNullOrWhiteSpace(databaseName))
        return;
    // put the database name in the orchestrator connection string
    // we use a SqlConnectionStringBuilder to avoid having to parse the connection string
    SqlConnectionStringBuilder sqlConnectionStringBuilder = new SqlConnectionStringBuilder(orchestrator.Provider.ConnectionString);
    sqlConnectionStringBuilder.InitialCatalog = databaseName;
    orchestrator.Provider.ConnectionString = sqlConnectionStringBuilder.ConnectionString;
    string sql = "SELECT name FROM sys.tables WHERE PATINDEX('%scope_%', name) = 0 AND PATINDEX('%sys%', name) = 0";
    List<string> tableList = new List<string>();
    SqlConnection myConnection = new SqlConnection(orchestrator.Provider.ConnectionString);
    myConnection.Open();
    using (SqlCommand command = new(sql, myConnection))
    {
        using SqlDataReader reader = command.ExecuteReader();
        while (reader.Read())
        {
            tableList.Add(reader.GetString(0));
        }
    }
    string[] tables = tableList.ToArray();
    orchestrator.Setup = new SyncSetup(tables);

    string contentRootPath = (string)AppDomain.CurrentDomain.GetData("ContentRootPath");
    orchestrator.Options.SnapshotsDirectory = Path.Combine(contentRootPath, "Snapshots", databaseName);
}

/// <summary>
/// This Get handler is optional. It allows you to see the configuration hosted on the server
/// The configuration is shown only if Environmenent == Development
/// </summary>
[HttpGet]
public async Task Get() => await webServerManager.HandleRequestAsync(this.HttpContext);

```
On the client
```
static SyncAgent ConfigureSyncAgent(bool createCloudSnapshot)
{
    SqliteConnectionStringBuilder builder = new SqliteConnectionStringBuilder(SqliteConnectionString);
    SqliteSyncProvider clientProvider = new SqliteSyncProvider(builder);

    string serviceUri = $"{SyncApi}?databaseName={DatabaseName}&createSnapshot={createCloudSnapshot}";

    WebClientOrchestrator webClientOrchestrator = new WebClientOrchestrator(serviceUri);

    webClientOrchestrator.HttpClient.Timeout = TimeSpan.FromMinutes(20);

    SyncOptions syncOptions = new SyncOptions();
    syncOptions.BatchDirectory = Path.Combine(AppDataFolder, "Snapshots");
    syncOptions.BatchSize = 2000; // ~ 5mb https://github.com/Mimetis/Dotmim.Sync/issues/368#issuecomment-739430229
    syncOptions.CleanMetadatas = true; // clears local tracking tables after successful sync
    syncOptions.UseBulkOperations = true;

    SyncAgent syncAgent = new SyncAgent(clientProvider, webClientOrchestrator, syncOptions);
    return syncAgent;
}

I'll upgrade to 0.6.2 again and post the stack trace here.
```

**Comment by VagueGit** [comment]
I thought maybe it was because I was upgrading from a beta so I uninstall DMS then a clean install of 0.6.2. But it didn't fix it. Now I have rolled back to 0.6.1, but sync on the client throws `Could not load file or assembly 'Utf8Json, Version=1.3.7.0, Culture=neutral, PublicKeyToken=8a73d3ba7e392e27'. The system cannot find the file specified.`
Stack trace
```
   at Dotmim.Sync.Web.Client.HttpRequestHandler.<ProcessRequestAsync>d__12`1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Dotmim.Sync.Web.Client.WebClientOrchestrator.<ApplyThenGetChangesAsync>d__32.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Dotmim.Sync.SyncAgent.<SynchronizeAsync>d__51.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Wbl.Services.DataService.<SyncNowAsync>d__30.MoveNext() in C:\Repos\FREE\Wbl\Services\DataService.cs:line 119
```
There is no error on the web server.

**Comment by Mimetis** [comment]
Try to install the utf8json package

**Comment by VagueGit** [comment]
Where does the utf8json package come from?

**Comment by Mimetis** [comment]
https://www.nuget.org/packages/Utf8Json


Last version of DMS is using it, replacing json.net

**Comment by VagueGit** [comment]
I would have thought DMS would install any dependencies.

I've installed Utf8Json nuget on the client. Still getting `setup does not contains any table`

**Comment by VagueGit** [comment]
I think I've found where the problem was coming from. We started with a SQLite database create using `DMS 0.6.1-beta-0504`. There are no schema changes between that version of the SQLite database and the version used when we upgraded to `DMS 0.6.2`. However once I allowed `DMS 0.6.2` to create a new SQLite database, DMS was able to sync without error.

So there my be some incompatibility between `DMS 0.6.1-beta-0504` stuff and `DMS 0.6.2`

**Comment by Mimetis** [comment]
This error should not happened actually, even when upgrading version.
Will look into it.

>Since we are working on mobile app, that would mean that after each update we would have to re-create local database and of course loose all data.

No, even if it's failing at some point (and it should not, I agree) you still can use the methods from the SDK to recreate the setup without losing anything

That being said, If you have a repro sample, it could help a lot to figure out what's happening here.

**Comment by MaTToX3** [comment]
> No, even if it's failing at some point (and it should not, I agree) you still can use the methods from the SDK to recreate the setup without losing anything

It does not even reach the server when syncing so it would mean I would have to re-create the setup from the client side but this sounds wrong to me. Or did I misunderstood something?

Regarding the sample, I will take a look at what can I provide.

**Comment by Mimetis** [comment]
I guess I have found the issue.
It's resolved in last package `v0.7.2`
Let me know if you still facing some issue

**Comment by VagueGit** [comment]
Confirmed fixed. Thank you @Mimetis
